lms/static/scripts/frontend_apps/components/dashboard/OrderableActivityTable.tsx

```diff
@@ -0,0 +1,70 @@
+import type { DataTableProps } from '@hypothesis/frontend-shared';
+import { DataTable } from '@hypothesis/frontend-shared';
+import { useOrderedRows } from '@hypothesis/frontend-shared';
+import type { OrderDirection } from '@hypothesis/frontend-shared/lib/types';
+import { useMemo, useState } from 'preact/hooks';
+
+import type { BaseDashboardStats } from '../../api-types';
+
+export type OrderableActivityTableProps<T extends BaseDashboardStats> = Pick<
+  DataTableProps<T>,
+  'emptyMessage' | 'rows' | 'renderItem' | 'loading' | 'title'
+> & {
+  columnNames: Partial<Record<keyof T, string>>;
+  initialOrderField: keyof T;
+};
+
+const statsColumns: ReadonlyArray<string> = [
+  'last_activity',
+  'annotations',
+  'replies',
+];
+
+export default function OrderableActivityTable<T extends BaseDashboardStats>({
+  initialOrderField,
+  rows,
+  columnNames,
+  ...restOfTableProps
+}: OrderableActivityTableProps<T>) {
+  const [order, setOrder] = useState<NonNullable<DataTableProps<T>['order']>>({
+    field: initialOrderField,
+    direction: 'ascending',
+  });
+  const orderedRows = useOrderedRows(rows, order);
+  const columns = useMemo(
+    () =>
+      Object.entries(columnNames).map(([field, label], index) => ({
+        field: field as keyof T,
+        label: label as string,
+        classes: index === 0 ? 'w-[60%]' : undefined,
+      })),
+    [columnNames],
+  );
+  const orderableColumns = useMemo(
+    () =>
+      (Object.keys(columnNames) as Array<keyof T>).reduce<
+        Partial<Record<keyof T, OrderDirection>>
+      >((acc, columnName) => {
+        // All stat columns should be ordered in descending order first
+        acc[columnName] =
+          typeof columnName === 'string' && statsColumns.includes(columnName)
```

Comment:
`columnName` has type `keyof T`, which translates into `string | number | symbol`.

Without the check for string type, TypeScript complains when passed to `Array<string>.includes()`.

This is related to the discussion here https://hypothes-is.slack.com/archives/C1M8NH76X/p1716819580379869, but I skipped this part from the conversation because it was easy to solve.